source/minimessage/api.rst

```diff
@@ -0,0 +1,215 @@
+
+   player.sendMessage(parsed);
+
+
+For more advanced uses, additional tag resolvers can be registered, which when given a tag name and arguments will produce a ``Tag`` instance. These are described in more detail below.
+
+Builder
+-------
+
+To make customizing MiniMessage easier, we provide a Builder. The specific methods on the builder are explained in the javadoc.
+
+.. code:: java
+
+    MiniMessage minimessage = MiniMessage.builder()
+        .tags(TagResolver.builder()
+          .resolver(StandardTags.color())
+          .resolver(StandardTags.decorations())
+          .resolver(this.additionalPlaceholders)
+         )
+        .build();
+
+.. tip::
+   
+   It's a good idea to initialize such a MiniMessage instance once, in a central location, and then use it for all your messages.
+   Exception being if you want to customize MiniMessage based on permissions of a user (for example, admins should be allowed to use color and decoration in the message, normal users not)
+
+Error handling
+--------------
+
+By default, MiniMessage will never throw an exception caused by user input. Instead, it will treat any invalid tags as normal text. ``MiniMessage.Builder#strict(true)`` mode will enable strict mode, 
+which throws exceptions on unclosed tags, but still will allow any improperly specified tags through.
+
+To capture information on why a parse may have failed, ``MiniMessage.Builder#debug(Consumer<String>)`` can be provided, which will accept debug logging for an input string.
+
+Tag resolvers
+^^^^^^^^^^^^^
+
+All tag resolution goes through tag resolvers. There is one global tag resolver, which describes the tags available through a `MiniMessage` instance, plus parse-specific resolvers which can provide additional input-specific tags.
+
+Tag resolvers are the binding between a name and arguments, and the logic to produce a ``Component`` contained in a ``Tag`` instance. They are composable so a ``TagResolver`` can produce any number of actual ``Tag`` instances. The tag name passed to resolvers will always be lower-cased, to ensure case-insensitve searches.
+
+MiniMessage has built-in resolver types that can be used for most use-cases, including custom tags and fixed-value placeholder-style tags. For single-tag resolvers, use the static factory methods in 
+``TagResolver`` and ``Placeholder``. To combine multiple resolvers, take a look at the tag resolver builder, ``TagResolver.builder()``.
+
+Placeholder resolvers are especially useful for simple cases:
+
+.. code:: java
+
+    MiniMessage.miniMessage().deserialize("<gray>Hello <name>", Placeholder.component("name", Component.text("TEST", NamedTextColor.RED)));
+    MiniMessage.miniMessage().deserialize("<gray>Hello <name>", Placeholder.unparsed("name", "TEST"));
+    TagResolver placeholders = TagResolver.resolver(Placeholder.parsed("name", "TEST"), Placeholder.parsed("name2", "TEST"));
```

Comment:
Worth bringing out the difference of `Placeholder.unparsed` and `Placeholder.parsed` directly here?